server/datastore/mysql/software.go

```diff
@@ -151,8 +175,10 @@ func getOrGenerateSoftwareIdDB(ctx context.Context, tx sqlx.ExtContext, s fleet.
 	var existingId []int64
 	if err := sqlx.SelectContext(ctx, tx,
 		&existingId,
-		`SELECT id FROM software WHERE name = ? AND version = ? AND source = ? AND bundle_identifier = ?`,
-		s.Name, s.Version, s.Source, s.BundleIdentifier,
+		"SELECT id FROM software "+
+			"WHERE name = ? AND version = ? AND source = ? AND `release` = ? AND "+
+			"vendor = ? AND arch = ? AND bundle_identifier = ?",
```

Comment:
Not a fan of the back quotes for string literals? :)

Reply:
Back quotes are needed due to "release" being a mysql keyword. Unless there's a better way.

Reply:
Ah, right, the alternative there is something like this:

```go
a := `hi Lucas, ` + "`" + `nice` + "`" + ` to meet you`
```